code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -139,6 +139,8 @@
 				update_tint()
 			if(G.vision_correction)
 				clear_fullscreen("nearsighted")
+				if(HAS_TRAIT_FROM_ONLY(src, TRAIT_ILLITERATE, FARSIGHT_TRAIT)) //ORBSTATION: remove illiteracy iff it's from the Farsighted quirk
```

Comment:
It is my understanding that calling `REMOVE_TRAIT` with a source should perform this check itself.
This check might actually make the logic _incorrect_.

If you: 
- take off your glasses
- gain illiteracy from a trauma
- put on your glasses
- cure the trauma
Is the trait from farsightedness still present?